dhcpv4/ztp/ztp.go

```diff
@@ -0,0 +1,139 @@
+package ztpv4
+
+import (
+	"bytes"
+	"errors"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// VendorData is optional data a particular vendor may or may not include
+// in the Vendor Class options.  All values are optional and will be zero
+// values if not found.
+type VendorData struct {
+	VendorName string
+	Model      string
+	Serial     string
+}
+
+var errVendorOptionMalformed = errors.New("malformed vendor option")
+
+// VendorDataV4 will try to parse dhcp4 options data looking for more specific
+// vendor data (like model, serial number, etc).  If the options are missing
+func VendorDataV4(packet *dhcpv4.DHCPv4) VendorData {
```

Comment:
this function should not have the redundant "v4" at the end as it's part of the dhcpv4 package already. A more appropriate name would be `ParseVendorData`, because that's what the function does